examples/tutorials/analysis-3d/energy_dependence_estimation.py

```diff
@@ -0,0 +1,266 @@
+
+energy_edges = [0.3, 1, 5, 10] * u.TeV
+
+######################################################################
+# Define the spectral and spatial models of interest. Here we utilise
+# a PowerLawSpectralModel and a GaussianSpatialModel to test the energy
+# dependent morphology component in each energy band. A standard 3D fit
+# is performed then the best fit model is utilised here for the parameters
+# in each model.
+
+spectral_model = PowerLawSpectralModel(
+    index=2.26, amplitude=2.58e-12 * u.Unit("cm-2 s-1 TeV-1"), reference=1.0 * u.TeV
+)
+
+spatial_model = GaussianSpatialModel(
+    lon_0=source_pos.l,
+    lat_0=source_pos.b,
+    frame="galactic",
+    sigma=0.11 * u.deg,
+    e=0.8346,
+    phi=-2.914 * u.deg,
+)
+
+# Limit the search for the position on the spatial model
+spatial_model.lon_0.min = source_pos.galactic.l.deg - 0.8
+spatial_model.lon_0.max = source_pos.galactic.l.deg + 0.8
+spatial_model.lat_0.min = source_pos.galactic.b.deg - 0.8
+spatial_model.lat_0.max = source_pos.galactic.b.deg + 0.8
+
+model = SkyModel(
+    spatial_model=spatial_model, spectral_model=spectral_model, name="MSH1552"
+)
+
+######################################################################
+# Run Estimator
+# -------------
+#
+# Now we can run the energy dependent estimation and explore the results.
+#
+# We start with the initial hypothesis, in which the source is added
+# in to compare with the background. We define here which parameters we
+# wish to use to test the energy dependence. To test the energy dependence
+# it is advised to let the position and extension parameter to be free
+# such that these can be used to fit the spatial model in each band.
+
+model.spatial_model.lon_0.frozen = False
+model.spatial_model.lat_0.frozen = False
+model.spatial_model.sigma.frozen = False
+
+model.spectral_model.amplitude.frozen = False
+model.spectral_model.index.frozen = True
+
+datasets.models = model
+
+estimator = EnergyDependenceEstimator(energy_edges=energy_edges, source="MSH1552")
+
+######################################################################
+# Show the results tables
+# -----------------------
+#
+# The results of the source signal above the background in each energy bin.
+# -----------------------------------------------
+#
+
+estimator = EnergyDependenceEstimator(energy_edges=energy_edges, source="MSH1552")
+result_bkg_src = estimator.estimate_source_significance(datasets)
+Table(result_bkg_src)
+
+######################################################################
+# The results for testing energy dependence
+# -----------------------------------------
+
+results = estimator.run(datasets)
+ts = results["energy_dependence"]["delta_ts"]
+df = results["energy_dependence"]["df"]
+sigma = ts_to_sigma(ts, df=df)
+print(f"The delta_ts for the energy-dependent study is {ts:.3f}.")
+print("")
+print(f"Converting this to a significance gives {sigma:.3f} \u03C3")
+table = Table(results["energy_dependence"]["result"])
```

Comment:
If `table` is used below (see next comment), maybe it would be wise to renamed it to a more specific name, i.e. `results_table`.